lib/github/markups.rb

```diff
@@ -52,3 +52,5 @@
 )
 
 command(::GitHub::Markups::MARKUP_POD, :pod2html, /pod/, ["Pod"], "pod")
+
+command(::GitHub::Markups::MARKUP_MANPAGE, :man2html, /(?:[1-9][a-z]*|0p|n|man|mdoc)(?:\.in)?/, ["Roff"], "manpage")
```

Comment:
Just noticed this regexp doesn't catch extensions with underscores or trailing digits:

* [`3c_db`](https://github.com/search?q=extension%3A3c_db+NOT+nothack&type=Code)
* [`3X11`](https://github.com/search?q=extension%3A3x11+NOT+nothack&type=Code)

To fix that, use:

~~~regexp
/(?:[1-9](?![1-9])[a-z_0-9]*|0p|n|man|mdoc)(?:\.in)?/
~~~

Reply:
Hi @Alhadis, Is your intent to allow `.40` and `.7_up_`? How about

    /
        (?:
            [1-9] (?:
                [a-z] [a-z\d]* (?:
                    _ [a-z\d]+
                )*
            )? |
            0p | n | man | mdoc
        )
        (?:\.in)?
        /

though it might need adjusting for Ruby's regexp syntax.

Reply:
@Alhadis My attention is going to be diverted elsewhere for a bit but feel free to push that change up to this branch

Reply:
Don't have access, I'm afraid...
~~~
ERROR: Permission to github/markup.git denied to Alhadis.
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
~~~

Reply:
Oof OK, if you've got it done already push it somewhere it'll let you and I'll cherry-pick it, otherwise I'll try and pick this off soon

Reply:
Oh, the commit I pushed was just

~~~diff
commit 61278c1be71b6c4f2c2f9ab007558c2ffd327698
Author: Alhadis <gardnerjohng@gmail.com>
Date:   Tue Nov 6 04:22:25 2018 +1100

    Match extensions that contain underscores or digits

diff --git a/lib/github/markups.rb b/lib/github/markups.rb
index 3143299..21f9f99 100644
--- a/lib/github/markups.rb
+++ b/lib/github/markups.rb
@@ -53,4 +53,4 @@ command(
 
 command(::GitHub::Markups::MARKUP_POD, :pod2html, /pod/, ["Pod"], "pod")
 
-command(::GitHub::Markups::MARKUP_MANPAGE, :man2html, /(?:[1-9][a-z]*|0p|n|man|mdoc)(?:\.in)?/, ["Roff"], "manpage")
+command(::GitHub::Markups::MARKUP_MANPAGE, :man2html, /(?:[1-9](?![0-9])[a-z_0-9]*|0p|n|man|mdoc)(?:\.in)?/, ["Roff"], "manpage")
~~~

I'll fork it so it's online where you can cherry-pick it. :)

I'm busy working on the ManpageStrategy which is so-far-so-good!

Reply:
Pushed as https://github.com/Cutlery-Drawer/markup/commit/61278c1be71b6c4f2c2f9ab007558c2ffd327698 👍 Cherry-pick away. =)

Reply:
@Alhadis You might want to make this into a [Suggested Change](https://twitter.com/github/status/1053360435074949120).

Reply:
Damn. Had no idea that feature existed.

👍

Reply:
It's new within the past couple months :) Applied!

Reply:
This thinks foo.7z is a man-page section. https://en.wikipedia.org/wiki/7z_(file_format)
Skimming https://en.wikipedia.org/wiki/List_of_filename_extensions_(alphabetical)#0%E2%80%939, I also notice

- readme.1st
- mobile-phone-video.3gp
- forth-source.4th

Is the regexp used ignoring case? My reading of that Ruby code is case is not ignored.

Reply:
@RalphCorderoy Linguist ignores binary file formats, so we're safe in regard to `.7z` and `.3gp` files.